File: frontend/src/api/skillmatrix.ts
/**
 * スキルマトリクス API クライアント
 *
 * エンジニアごとのスキル保有状況をマトリクス形式で取得する。
 */
import apiClient from './client'
import type { EmployeeListItem } from './employees'

export interface SkillInfo {
  id: string
  name: string
}

export interface SkillMatrixCategory {
  id: string
  name_ja: string
  skills: SkillInfo[]
}

export interface EngineerSkillEntry {
  skill_id: string
  level: number | null
  years: number | null
}

export interface EngineerRow {
  employee: EmployeeListItem
  skills: Record<string, EngineerSkillEntry>
}

export interface SkillMatrixResponse {
  categories: SkillMatrixCategory[]
  engineers: EngineerRow[]
}

export interface SkillMatrixParams {
  location?: string
  level_min?: number
  category_id?: string
  free_only?: boolean
  search?: string
}

export const skillMatrixApi = {
  get: (params?: SkillMatrixParams) =>
    apiClient.get<SkillMatrixResponse>('/skills/matrix', { params }).then(r => r.data),
}

File: backend/app/schemas/skillmatrix.py
"""スキルマトリクススキーマ"""
from pydantic import BaseModel
from app.schemas.employee import EmployeeListItem


class SkillInfo(BaseModel):
    id: str
    name: str   # skill.name_ja ?? skill.name


class EngineerSkillEntry(BaseModel):
    skill_id: str
    level: int | None = None
    years: float | None = None


class SkillMatrixCategory(BaseModel):
    id: str
    name_ja: str
    skills: list[SkillInfo]


class EngineerRow(BaseModel):
    employee: EmployeeListItem
    skills: dict[str, EngineerSkillEntry]   # skill_id -> entry


class SkillMatrixResponse(BaseModel):
    categories: list[SkillMatrixCategory]
    engineers: list[EngineerRow]
